types/react-native-sortable-list/index.d.ts

```diff
@@ -1,150 +1,158 @@
      * the end of the list that will trigger autoscrolling. Defaults to 60.
      */
-    autoscrollAreaSize?: number
+    autoscrollAreaSize?: number;
 
     /**
      * determines time delay in ms before pressed row becomes active. Defaults to 200 ms.
      */
-    rowActivationTime?: number
+    rowActivationTime?: number;
 
     /**
      * A RefreshControl that works the same way as a ScrollView's refreshControl.
      */
-    refreshControl?: JSX.Element
+    refreshControl?: ReactElement;
 
     /**
      * Takes a row key, row index, data entry from the data source and its statuses disabled, active and should
      * return a renderable component to be rendered as the row. The child component will receive a method called
      * toggleRowActive (only if manuallyActivateRows={true}) to manually activate the row. Useful if you have
      * multiple touch responders in your view.
      */
-    renderRow: (props: RowProps) => JSX.Element
+    renderRow: (props: RowProps<T, K>) => ReactElement | null;
 
     /**
      * Renders returned component at the top of the list.
      */
-    renderHeader?: () => JSX.Element
+    renderHeader?: () => ReactElement;
 
     /**
      * Renders returned component at the bottom of the list.
      */
-    renderFooter?: () => JSX.Element
+    renderFooter?: () => ReactElement;
 
     /**
      * Called when rows were reordered, takes an array of rows keys of the next rows order.
      */
-    onChangeOrder?: (nextOrder: DataKey[]) => void
+    onChangeOrder?: (nextOrder: K[]) => void;
 
     /**
      * Called when a row was activated (user long tapped).
      */
-    onActivateRow?: (key: DataKey) => void
+    onActivateRow?: (key: K) => void;
 
     /**
      * Called when the active row was released.
      */
-    onReleaseRow?: (key: DataKey) => void
+    onReleaseRow?: (key: K, currentOrder: K[]) => void;
 
     /**
      * Called when a row was pressed.
      */
-    onPressRow?: (key: DataKey) => void
+    onPressRow?: (key: K) => void;
 }
 
-interface SortableListStatic extends React.ClassicComponentClass<SortableListProps> {}
+export default class SortableList<T, K> extends Component<SortableListProps<T, K>> {
+    /**
+     * scrolls by a given y offset, either immediately or with a smooth animation
+     */
+    scrollBy({dx, dy, animated}: { dx?: number; dy?: number; animated?: boolean }): void;
+
+    /**
+     * scrolls to a given y offset, either immediately or with a smooth animation
+     */
+    scrollTo({x, y, animated}: { x?: number; y?: number; animated?: boolean }): void;
 
-declare var SortableList: SortableListStatic;
-declare type SortableList = SortableListStatic;
+    /**
+     * scrolls to a given row key, either immediately or with a smooth animation
+     */
+    scrollToRowKey({key, animated}: { key?: K; animated?: boolean }): void;
+}
 
-export default SortableList;
+export { SortableList, RowProps };
```

Comment:
This appears to be wrong, according to https://unpkg.com/react-native-sortable-list@0.0.23/index.js. `export default` was correct.

Reply:
To @sivolobov’s point earlier, you can keep the `export { RowProps }` to avoid the implicit exports, but `SortableList` should be a default export.